crates/ruff_python_trivia/src/line_width.rs

```diff
@@ -0,0 +1,103 @@
+use ruff_cache::{CacheKey, CacheKeyHasher};
+use std::error::Error;
+use std::fmt::Debug;
+use std::hash::Hasher;
+use std::num::ParseIntError;
+use std::str::FromStr;
+
+/// The length of a line of text that is considered too long.
+///
+/// The allowed range of values is 1..=320
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub struct LineLength(u16);
```

Comment:
My preference is to keep them separate. The `LineLength` in the settings builds our API to users (and the configuration), the one in the formatter doesn't need to have the same constraints. The way I would avoid too much duplicate code is to only implement the parsing on the `LineLength` in the settings crate (and remove it from `ruff_formatter`). It is no longer needed on the formatter because we can directly convert between the two without needing any conversion.

Reply:
I can make that change on Monday